go/engine/pgp_encrypt.go

```diff
@@ -190,20 +190,42 @@ func (e *PGPEncrypt) loadSelfKey() (*libkb.PGPKeyBundle, error) {
 }
 
 func (e *PGPEncrypt) verifyUsers(ctx *Context, assertions []string, loggedIn bool) ([]string, error) {
+	// Get current user for tracking. Ignore errors, but look out for
+	// me == nil, e.g. when user is not logged in (pgp encrypt can be
+	// used without Keybase account).
+	me, _ := libkb.LoadMe(libkb.NewLoadUserArg(e.G()))
+
 	var names []string
 	for _, userAssert := range assertions {
 		arg := keybase1.Identify2Arg{
 			UserAssertion: userAssert,
-			Reason: keybase1.IdentifyReason{
-				Type: keybase1.IdentifyReasonType_ENCRYPT,
-			},
-			AlwaysBlock: true,
+			AlwaysBlock:   true,
+			NeedProofSet:  true,
+		}
+		topts := keybase1.TrackOptions{
+			LocalOnly: me == nil,
 		}
-		eng := NewResolveThenIdentify2(e.G(), &arg)
-		if err := RunEngine(eng, ctx); err != nil {
+		ieng := NewResolveThenIdentify2WithTrack(e.G(), &arg, topts)
+		if err := RunEngine(ieng, ctx); err != nil {
 			return nil, libkb.IdentifyFailedError{Assertion: userAssert, Reason: err.Error()}
 		}
-		res := eng.Result()
+
+		res := ieng.Result()
+		confirmResult := ieng.ConfirmResult()
+		if !confirmResult.IdentityConfirmed {
+			return nil, libkb.IdentifyFailedError{Assertion: userAssert, Reason: "Not confirmed by user."}
+		}
+
+		if me != nil && confirmResult.RemoteConfirmed {
+			targ := &TrackTokenArg{
+				Token:   ieng.TrackToken(),
+				Me:      me,
+				Options: keybase1.TrackOptions{},
+			}
+			teng := NewTrackToken(targ, e.G())
+			RunEngine(teng, ctx)
```

Comment:
I might need to take another look then, last time I was working on it we thought that posting track/follow statement is just not there at all during `pgp encrypt`.